.github/workflows/test_and_release.yml

```diff
@@ -56,19 +56,37 @@ jobs:
       with:
         python-version: ${{ matrix.python-version }}
 
+    - name: Install trame-vtk.js
+      run: |
+        mkdir -p ./trame_vtk/modules/common/serve
+        curl https://unpkg.com/vue-vtk-js@3.1.6 -Lo ./trame_vtk/modules/common/serve/trame-vtk.js
+
     - name: Install and Run Tests
       run: |
         pip install .
         pip install -r tests/requirements.txt
+        pip uninstall vtk -y
+        pip install --extra-index-url https://wheels.vtk.org vtk-osmesa
         # Run the tests with coverage so we get a coverage report too
         pip install coverage
-        coverage run --source . -m pytest . --browser=firefox
+        coverage run --source . -m pytest . --firefox --headless --html=report-${{ matrix.config.name }}-${{ matrix.python-version }}.html
         # Print the coverage report
         coverage report -m
```

Comment:
I recommend separating the testing into a separate `run` step. This makes it easier to parse the logs when errors inevitably happen in either the installation stage or testing stage